week1/reposts/thalesmg/Main.hs

```diff
@@ -0,0 +1,36 @@
+import System.Random
+import Control.Monad.State
+
+nPrizes = 7
+totalReposts = 1000000
+nSims = 10000
+
+simulate :: Int -> Int -> State StdGen [Int]
+simulate n tr =
+  let
+    loto :: State StdGen Int
+    loto = state $ randomR (1, tr)
+    sims :: State StdGen [Int]
+    sims = replicateM n loto
```

Comment:
> seven prizes will be given to seven randomly chosen people among those who have reposted a certain post

Your solution doesn't ensure that chosen people would be different, right? It should but it doesn't matter much regarding the answer so let's pretend it's okay :)